frontend/pages/review.tsx

```diff
@@ -35,29 +37,88 @@ function ErrorWrapper({ message }: { message: string | undefined }) {
+  const { approvals, isApprovalsLoading, isApprovalsError, mutateApprovals } = useListApprovals(category, filter)
+
+  const error = MultipleErrorWrapper(
+    `Unable to load approval page`,
+    {
+      isApprovalsError,
+    },
+    ErrorWrapper
+  )
+  if (error) return error
+
+  const getUploadCategory = (approvalCategory: ApprovalCategory) =>
+    approvalCategory === 'Upload' ? 'models' : 'deployments'
+
+  if (isApprovalsLoading || !approvals) {
+    return <Paper sx={{ mt: 2, mb: 2 }} />
+  }
+  return (
+    <Paper sx={{ mt: 2, mb: 2, pb: 2 }}>
+      <Typography sx={{ p: 3 }} variant='h4'>
+        {category === 'Upload' ? 'Models' : 'Deployments'}
+      </Typography>
+      {approvals.map((approval: Approval) => (
+        <ApprovalItem
+          approval={approval}
+          category={approval.approvalCategory}
+          version={approval.version as VersionDoc}
+          key={approval._id}
+          mutateApprovals={mutateApprovals}
+        />
+      ))}
+
+      {approvals.length === 0 && (
+        <EmptyBlob text={`All done! No ${getUploadCategory(category)} are waiting for approval.`} />
+      )}
+    </Paper>
+  )
+}
+
+type ApprovalItemProps = {
+  approval: Approval // changing this from 'any' to 'Approval' causes errors
+  category: string
+  version: VersionDoc
+
+  mutateApprovals: () => void
+}
+
+function ApprovalItem({ approval, category, mutateApprovals }: ApprovalItemProps) {
   const [open, setOpen] = useState(false)
   const [choice, setChoice] = useState('')
-  const [approval, setApproval] = useState<Approval | undefined>(undefined)
+  //const [getapproval, setGetApproval] = useState<Approval | undefined>(undefined)
   const [approvalModalText, setApprovalModalText] = useState('')
   const [approvalModalTitle, setApprovalModalTitle] = useState('')
   const [showAlert, setShowAlert] = useState(false)
   const [errorMessage, setErrorMessage] = useState('')
-  const theme = useTheme()
 
-  const { approvals, isApprovalsLoading, isApprovalsError, mutateApprovals } = useListApprovals(category, filter)
   const { mutateNumApprovals } = useGetNumApprovals()
+  const theme = useTheme()
 
-  const managerStyling = {
-    mb: 2,
-    borderLeft: '.3rem solid #283593',
-    p: 2,
-    backgroundColor: theme.palette.container.main,
+  const { schema, isSchemaLoading, isSchemaError } = useGetSchema(
+    ((approval.version as VersionDoc).model as ModelDoc)?.schemaRef ||
+      ((approval.deployment as DeploymentDoc)?.model as ModelDoc)?.schemaRef
+  )
+
+  const reviewer = schema?.schema.properties.contacts.properties.reviewer.title
+  if (isSchemaError) {
+    return <ErrorWrapper message={isSchemaError.info.message} />
   }
-  const reviewerStyling = {
-    mb: 2,
-    borderLeft: '.3rem solid #de3c30',
-    p: 2,
-    backgroundColor: theme.palette.container.main,
+
+  if (isSchemaLoading) {
+    return null
+  }
+
+  function canApprove() {
```

Comment:
This could be re-rwitten using `useMemo`:
```
  const canApprove = useMemo(() => {
    if (!approval.version || approval.approvalType === 'Reviewer') {
      return true
    }
    return (
      approval.approvalType === 'Manager' &&
      (approval.version as VersionDoc).reviewerApproved === ApprovalStates.Accepted
    )
  }, [approval])
```